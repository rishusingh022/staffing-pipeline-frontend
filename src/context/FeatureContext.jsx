import React, { createContext, useState } from 'react';
import { PropTypes } from 'prop-types';

export const FeatureContext = createContext();

export function FeatureProvider({ children }) {
  const [userInfo, setUserInfo] = useState({
    featureAccess: [],
  });

  return <FeatureContext.Provider value={{ userInfo, setUserInfo }}>{children}</FeatureContext.Provider>;
}
FeatureProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
