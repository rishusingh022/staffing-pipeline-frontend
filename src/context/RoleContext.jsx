import React, { createContext, useState } from 'react';
import { PropTypes } from 'prop-types';

export const RoleContext = createContext();

export function RoleProvider({ children }) {
  const [userInfo, setUserInfo] = useState({});

  return <RoleContext.Provider value={{ userInfo, setUserInfo }}>{children}</RoleContext.Provider>;
}
RoleProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
