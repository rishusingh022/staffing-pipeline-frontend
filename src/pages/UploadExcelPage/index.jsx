import React from 'react';
import Header from '../../components/Header';
import Button from '../../components/Button';
import PageLoader from '../../components/Spinner';

function UploadExcelPage() {
  const [isLoading] = React.useState(false);
  const [file, setFile] = React.useState(null);
  const handleFileSelect = event => {
    setFile(event.target.files[0]);
  };
  const handleUpload = () => {
    if (file) {
      console.log('file selected');
    } else {
      console.log('file not selected');
    }
  };

  return isLoading ? (
    <div>
      <PageLoader />
    </div>
  ) : (
    <div>
      <Header hasNav />
      <div className="w-full flex flex-col items-center">
        <p className="text-3xl text-center p-10">Upload Staffing Excel</p>
        <div className="bg-[#f8f8ee] w-4/5 p-8 rounded-xl border-2 border-dotted border-[#ad8a09]">
          <p className="text-lg pb-4">Rules to upload excel: </p>
          <ul className="list-disc list-inside">
            <li>
              Excel file must be in <b>.xlsx</b> format
            </li>
            <li>
              Excel file must have atleast 4 columns: <b>UserId, EngagmentId, AssignmentStartDate</b> and{' '}
              <b>AssignmentEndDate.</b>
            </li>
            <li>
              Excel file must have <b>1 header</b> row
            </li>
            <li>
              Excel file must have at least <b>1 data</b> row
            </li>
            <li>
              Excel file must <b>not</b> have any <b>empty</b> cells
            </li>
          </ul>
        </div>
        {file && (
          <div className="flex gap-4 p-4">
            <p className="font-bold">Selected File: </p>
            <p>{file.name}</p>
          </div>
        )}
        <div className="flex gap-4 py-4">
          <label
            htmlFor="excelUpload"
            className="bg-[#051b2c] text-white px-3 py-2 text-sm flex items-center cursor-pointer">
            Select Excel
          </label>
          <input
            onChange={handleFileSelect}
            id="excelUpload"
            className="hidden"
            type={'file'}
            accept=".csv, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"></input>
          <Button buttonText="Upload" handleClick={handleUpload} />
        </div>
      </div>
    </div>
  );
}

export default UploadExcelPage;
