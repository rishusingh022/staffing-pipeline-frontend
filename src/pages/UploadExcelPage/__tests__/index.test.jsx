import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { FeatureProvider } from '../../../context/FeatureContext';
import UploadExcelPage from '..';

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

jest.mock('@okta/okta-react', () => ({
  useOktaAuth: () => ({
    authState: {
      isAuthenticated: true,
      idToken: {
        claims: {
          email: '',
          name: '',
        },
      },
    },
  }),
}));

describe('Upload Excel Page', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(
      <FeatureProvider>
        <UploadExcelPage />
      </FeatureProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should show Select File button', () => {
    render(
      <FeatureProvider>
        <UploadExcelPage />
      </FeatureProvider>
    );
    const selectButton = screen.getByText('Select Excel');
    expect(selectButton).toBeTruthy();
  });
  it('should show Upload button', () => {
    render(
      <FeatureProvider>
        <UploadExcelPage />
      </FeatureProvider>
    );
    const uploadButton = screen.getByText('Upload');
    expect(uploadButton).toBeTruthy();
  });
});
