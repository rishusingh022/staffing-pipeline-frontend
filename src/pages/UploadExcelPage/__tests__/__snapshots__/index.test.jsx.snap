// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Excel Page should render without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="font-light lg:h-[88px] h-[120px] flex-shrink-0 text-white flex justify-center bg-deepBlue header-container relative items-center"
    >
      <img
        alt="logo"
        class="h-[70px] w-[155px] aspect-auto self-end pl-0 pt-0 pb-[5px] absolute xl:left-44 left-10 top-4"
        src="McK_Logo.png"
      />
      <div
        class="h-full  flex box-border"
      >
        <div
          class="grid grid-cols-4 gap-10 items-end"
        >
          <button
            class="text-[white] hover:text-cyan hover:border-b-[5px] hover:border-b-cyan hover:border-solid pt-0 pb-[15px] hover:pb-2.5 px-[5px]"
          >
            People
          </button>
          <button
            class="text-[white] hover:text-cyan hover:border-b-[5px] hover:border-b-cyan hover:border-solid pt-0 pb-[15px] hover:pb-2.5 px-[5px]"
          >
            Case Studies
          </button>
        </div>
      </div>
      <button
        class="absolute top-7 right-[5%] border-[1px] py-1 px-3 hover:text-cyan hover:border-cyan"
      >
        Logout
      </button>
    </div>
    <div
      class="w-full flex flex-col items-center"
    >
      <p
        class="text-3xl text-center p-10"
      >
        Upload Staffing Excel
      </p>
      <div
        class="bg-[#f8f8ee] w-4/5 p-8 rounded-xl border-2 border-dotted border-[#ad8a09]"
      >
        <p
          class="text-lg pb-4"
        >
          Rules to upload excel: 
        </p>
        <ul
          class="list-disc list-inside"
        >
          <li>
            Excel file must be in 
            <b>
              .xlsx
            </b>
             format
          </li>
          <li>
            Excel file must have atleast 4 columns: 
            <b>
              UserId, EngagmentId, AssignmentStartDate
            </b>
             and 
            <b>
              AssignmentEndDate.
            </b>
          </li>
          <li>
            Excel file must have 
            <b>
              1 header
            </b>
             row
          </li>
          <li>
            Excel file must have at least 
            <b>
              1 data
            </b>
             row
          </li>
          <li>
            Excel file must 
            <b>
              not
            </b>
             have any 
            <b>
              empty
            </b>
             cells
          </li>
        </ul>
      </div>
      <div
        class="flex gap-4 py-4"
      >
        <label
          class="bg-[#051b2c] text-white px-3 py-2 text-sm flex items-center cursor-pointer"
          for="excelUpload"
        >
          Select Excel
        </label>
        <input
          accept=".csv, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet, application/vnd.ms-excel"
          class="hidden"
          id="excelUpload"
          type="file"
        />
        <button
          class="button"
          data-testid="button"
        >
          Upload
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
